applications/MeshingApplication/custom_utilities/mmg_utilities.h

```diff
@@ -0,0 +1,645 @@
+     * @param Metric This array contains the components of the metric vector
+     * @param NodeId The id of the node
+     */
+    void SetMetricVector(
+        const array_1d<double, Dimension>& Metric,
+        const IndexType NodeId
+        );
+
+    /**
+     * @brief This function is used to compute the Hessian metric tensor, note that when using the Hessian, more than one metric can be defined simultaneously, so in consecuence we need to define the elipsoid which defines the volume of maximal intersection
+     * @param Metric This array contains the components of the metric tensor in the MMG defined order
+     * @param NodeId The id of the node
+     */
+    void SetMetricTensor(
+        const TensorArrayType& Metric,
+        const IndexType NodeId
+        );
+
+    ///@}
+    ///@name Access
+    ///@{
+
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const // override
+    {
+        return "MmgUtilities";
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const // override
+    {
+        rOStream << "MmgUtilities";
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const // override
+    {
+    }
+
+protected:
+
+    ///@name Protected Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+    ///@}
+
+private:
+    ///@name Static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+    ///@}
+    ///@name Private  Access
+    ///@{
+
+
+    ///@}
+    ///@name Private Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Un accessible methods
+    ///@{
+
+//     /// Assignment operator.
+//     MmgUtilities& operator=(MmgUtilities const& rOther);
+
+//     /// Copy constructor.
+//     MmgUtilities(MmgUtilities const& rOther);
+
+    ///@}
+
+};// class MmgUtilities
+///@}
+
+///@name Type Definitions
+///@{
+
+
+///@}
+///@name Input and output
+///@{
+
+/// input stream function
+template<MMGLibray TMMGLibray>
+inline std::istream& operator >> (std::istream& rIStream,
+                                  MmgUtilities<TMMGLibray>& rThis);
+
+/// output stream function
+template<MMGLibray TMMGLibray>
+inline std::ostream& operator << (std::ostream& rOStream,
+                                  const MmgUtilities<TMMGLibray>& rThis)
+{
+    rThis.PrintInfo(rOStream);
+    rOStream << std::endl;
+    rThis.PrintData(rOStream);
+
+    return rOStream;
+}
+
```

Comment:
You could do `extern template class MmgUtilities<...>;` here to avoid compilation overhead since you already instantiate explicitly.

Reply:
OK